### DISCUSSION TITLE
Unable to filter an array inside of a related model.

### DISCUSSION THREAD

**Discussion by alur2191** [prisma/prisma#11775] [Q&A] [answered]
I'm trying to check if the provided value exists inside of an array. I've been trying to figure this one out and from what I gathered, I have to use `has`. The array I'm trying to filter is inside of a related model. I tried looking for a solution, but couldn't find much on this subject. Am I doing something wrong? Is it at all possible to filter an array inside of a related model?

Here's my schema. Job and Company models are related, and inside Company we have a parking array.

```
model Company {
  id         Int       @id @default(autoincrement())
  name       String    @db.VarChar(200)
  state        String    @db.VarChar(30)
  parking    String[]
  ...
  createdAt  DateTime  @default(now())
  updated_at DateTime  @updatedAt
  @@map(name: "company")
}

model Job {
  id              Int      @id @default(autoincrement())
  type            String
  company         Company  @relation(fields: [company_id], references: [id])
  company_id      Int
  createdAt       DateTime @default(now())
  updated_at      DateTime @updatedAt
  UserJobs UserJobs[]
  @@map(name: "job")
}
```

Below, I'm trying to find many jobs which match various values. One of the values I'm trying to match is inside of an array in the related Company model. Here's what I tried:

```
     const jobs = await prisma.job.findMany({
            where: {
                AND: [
                    {
                        type: {
                            contains: req.body.type,
                        }
                    },
                    {
                        company: {
                            state: {
                                contains: req.body.state
                            }
                        }
                    },
                    ...
                      
                    {
                        company: {
                            parking: {
                                has: req.body.parkingState
                            }
                        }
                    }
                ]
            },
            include: {
                company: true,
            }
        })
```

Edit: removed most of the unrelated code, marked with `...`

**Comment by alur2191** [comment]
Bump.
I would really appreciate it if someone came through and told me if it's at all possible to filter through an array inside of a related model.
Thank you!

**Comment by chemicalkosek** [comment, marked as answer]
You would need to use relation filters. Have you tried using them?

https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#relation-filters
https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#relation-filters

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
